File: lobster-input-win/LobsterInput/Stores/RecordingResultStore.cs
using CommunityToolkit.Mvvm.ComponentModel;

namespace LobsterInput.Stores;

public sealed partial class RecordingResultStore : ObservableObject
{
    public static RecordingResultStore Instance { get; } = new();

    private RecordingResultStore() { }

    [ObservableProperty]
    private string _transcript = "";

    [ObservableProperty]
    private string _result = "";

    [ObservableProperty]
    private string? _errorMessage;

    public void Update(string transcript, string result, string? error)
    {
        Transcript = transcript;
        Result = result;
        ErrorMessage = error;
    }

    public void Clear()
    {
        Transcript = "";
        Result = "";
        ErrorMessage = null;
    }
}

File: lobster-input-ios/lobster-input-ios/Core/Audio/RecordingResultStore.swift
import Foundation
import Combine

@MainActor
final class RecordingResultStore: ObservableObject {

    static let shared = RecordingResultStore()
    private init() {}

    @Published var transcript: String = ""
    @Published var result: String = ""
    @Published var errorMessage: String?

    func update(transcript: String, result: String, error: String?) {
        self.transcript = transcript
        self.result = result
        self.errorMessage = error
    }

    func clear() {
        transcript = ""
        result = ""
        errorMessage = nil
    }
}
